platforms/documentation/docs/src/snippets/deprecationApi/deprecationExamples/kotlin/samplePlugin/src/main/java/org/gradle/sample/SamplePlugin.java

```diff
@@ -0,0 +1,48 @@
+/*
+ * Copyright 2025 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.sample;
+
+import org.gradle.api.Plugin;
+import org.gradle.api.Project;
+import org.gradle.api.problems.Problems;
+import org.gradle.api.problems.deprecation.source.ReportSource;
+
+import javax.inject.Inject;
+
+public abstract class SamplePlugin implements Plugin<Project> {
+
+    @Inject
+    public abstract Problems getProblems();
+
+    @Override
+    public void apply(Project project) {
+        getProblems().getDeprecationReporter().deprecatePlugin(
+            ReportSource.plugin("org.gradle.sample.plugin"),
+            "org.gradle.sample.plugin",
+            spec -> spec
+                .removedInVersion("2.0.0")
```

Comment:
🤔 This reads strange, because the plugin will change its identity. At the moment this reads as "the version is the primary thing that keep increasing, but we change the plugin id". Not sure how this can be better, so it's more of an observation

Reply:
Do you think this would be non-trivial when read on the user side?
I imagine something like this shown to the user:
> The plugin 'my.plugin.id' is deprecated, and will be replaced by plugin 'my-newer.plugin.id' from version "2.0.0"

I'm afraid I don't understand completely what you mean by "the version is the primary thing that keep increasing, but we change the plugin id". I feel that there are a few subtleties here, especially because we cannot define the replacement version.

Can you rephrase a bit your concern?